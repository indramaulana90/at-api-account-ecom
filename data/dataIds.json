[
	{
		"relativePath": "__tests__/v1/example/post/example/data.js",
		"ids": {
			"idUsers": 999999999
		}
	},
	{
		"relativePath": "__tests__/v1/order/get/order_parcel_detail/data.js",
		"ids": {
			"idOrders": 839640065,
			"idOrderParcels": 839641572,
			"idOrderParcelProducts": 839644779,
			"idBrands": "888888f39970108f39970108",
			"idBuyers": 839966387,
			"idUrbans": 839966387,
			"idInvoiceGroups": "888888f39970108f39970108",
			"idCart": "888888f39970314f39970314",
			"idDevice": "839966387",
			"idProducts": "888888f39970108f39970108",
			"idPhoneNumber": 839981005
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/workday_list/data.js",
		"ids": {
			"idWorkingDays": [
				841736409,
				841736511,
				841736535
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/workday_detail/data.js",
		"ids": {
			"idWorkingDays": [
				841802581,
				841802625
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/delete/workingDay/data.js",
		"ids": {
			"idWorkingDays": [
				841888623,
				841888635,
				841888672
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/patch/update_workingday/data.js",
		"ids": {
			"idWorkingDays": [
				841909018,
				841909045,
				841909072
			]
		}
	},
	{
		"relativePath": "__tests__/v1/order/post/cancel_order_parcel/data.js",
		"ids": {
			"idOrders": 842063620,
			"idOrderParcels": 842063620,
			"idOrderParcelProducts": 842063620,
			"idBrands": "888888f42063620f42063620",
			"idBuyers": 842063620,
			"idUrbans": 842063620,
			"idInvoiceGroups": "888888f42063620f42063620",
			"idCart": "888888f42063620f42063620",
			"idDevice": 842063620,
			"idProducts": "888888f42063620f42063620",
			"idPhoneNumber": 842063620
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/holiday_list/data.js",
		"ids": {
			"idHolidays": [
				841986121,
				841986145,
				841986166,
				841986181,
				841986198
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/holiday_detail/data.js",
		"ids": {
			"idHolidays": [
				841994725,
				841994752
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/delete/holidays/data.js",
		"ids": {
			"idHolidays": [
				841999606,
				841999633,
				841999652
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/post/holiday/data.js",
		"ids": {
			"idHolidays": 842147917
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/patch/holiday_update/data.js",
		"ids": {
			"idHolidays": [
				842318652,
				842318674,
				842318692,
				842321145
			]
		}
	},
	{
		"relativePath": "__tests__/v1/order/patch/update_order_status/data.js",
		"ids": {
			"idOrders": 842412325,
			"idOrderParcels": 842412325,
			"idOrderParcelProducts": 842412325,
			"idBrands": "888888f42412325f42412325",
			"idBuyers": 842412325,
			"idUrbans": 842412325,
			"idInvoiceGroups": "888888f42412325f42412325",
			"idCart": "888888f42412325f42412325",
			"idDevice": 842412325,
			"idProducts": "888888f42412325f42412325",
			"idPhoneNumber": 842412325
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/order_sellers_is_read/data.js",
		"ids": {
			"idOrders": 856578262,
			"idOrderSellers": 856578315
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/update_order_seller_failed/data.js",
		"ids": {
			"idOrders": 856642262,
			"idOrderSellers": 856642262,
			"idCart": "888888f56642550f56642550",
			"idDevice": 856642550,
			"idOrderBilling": 856642736,
			"idOrderSellerStatusHistoriesId": 856642736,
			"idOrderProduct": 856642736
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/update_order_seller_cancel/data.js",
		"ids": {
			"idOrders": 856664266,
			"idOrderSellers": 856664266
		}
	},
	{
		"relativePath": "__tests__/v1/warehouse/post/assign_product/data.js",
		"ids": {
			"idWarehouses": 862452416,
			"idSyncProduct": "62ec8be49c912a4015db7b41"
		}
	},
	{
		"relativePath": "__tests__/v1/warehouse/patch/warehouse_assign_product/data.js",
		"ids": {
			"idSyncProduct": "62ec8be49c912a4015db7b41"
		}
	}
]