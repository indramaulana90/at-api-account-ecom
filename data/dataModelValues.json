{
    "banners":{
        "name": "ATAPI ${params.atName} Banner Name",
        "header": "ATAPI${params.atName} Header",
        "description": "ATAPI ${params.atName} Description",
        "termAndCondition": "<p>• Berlaku untuk user dari semua tier</p><p>• Pembelian min 1.000.000 untuk SKU (SGM 300gr)</p><p>• Tidak berlaku kelipatan</p>"
    },
    "brands": {
        "name": "ATAPI ${params.atName} Brand Name",
        "code": "ATAPI${params.atName}Brand"
    },
    "buyers": {
        "name": "ATAPI ${params.atName} Buyers Name",
        "address": "ATAPI ${params.atName} Address"
    },
    "catalogues": {
        "name": "ATAPI ${params.atName} Catalogue Name",
        "code": "ATAPI${params.atName}Cata",
        "detail": "ATAPI ${params.atName} Catalogue Detail",
        "description": "ATAPI ${params.atName} Catalogue Description",
        "information": "ATAPI ${params.atName} Catalogue Information"
    },
    "catalogueCategories": {
        "name": "ATAPI ${params.atName} Category Name",
        "code": "ATAPI${params.atName}CCat"
    },
    "catalogueUnits": {
        "name": "ATAPI ${params.atName} Catalogue Unit Name"
    },
    "channels": {
        "name": "ATAPI ${params.atName} Channel Name",
        "externalId": "ATAPI${params.atName}Chan",
        "description": "ATAPI ${params.atName} Description"
    },
    "clusters": {
        "name": "ATAPI ${params.atName} Cluster Name",
        "externalId": "ATAPI${params.atName}Clus",
        "description": "ATAPI ${params.atName} Description"
    },
    "groups": {
        "name": "ATAPI ${params.atName} Group Name",
        "externalId": "ATAPI${params.atName}Grup",
        "description": "ATAPI ${params.atName} Description"
    },
    "invoiceGroups": {
        "name": "ATAPI ${params.atName} Invoice Group name",
        "code": "ATAPI${params.atName}InGr",
        "externalId": "ATAPI${params.atName}InGr"
    },
    "mssCatalogues": {
        "name": "ATAPI ${params.atName} MssCatalogues Name",
        "externalId": "ATAPI${params.atName}MSS"
    },
    "portfolios": {
        "name": "ATAPI ${params.atName} Portfolio Name",
        "externalId": "ATAPI${params.atName}Ptfl"
    },
    "products": {
        "name": "ATAPI ${params.atName} Product Name",
        "externalId": "ATAPI${params.atName}Pdct"
    },
    "stores": {
        "address": "ATAPI ${params.atName} Stores Address",
        "name": "ATAPI ${params.atName} Stores Name",
        "noteAddress": "ATAPI ${params.atName} Stores Note Address",
        "externalId": "ATAPI${params.atName}Strs"
    },
    "salesRep": {
        "name": "ATAPI ${params.atName} Users Sales fullname",
        "externalId": "ATAPI${params.atName}UsSl"
    },
    "salesTeamStoreChannels": {
        "name": "ATAPI ${params.atName} ST Store Channels Name"
    },
    "types": {
        "name": "ATAPI ${params.atName} Type Name",
        "externalId": "ATAPI${params.atName}Type",
        "description": "ATAPI ${params.atName} Description"
    },
    "urbans": {
        "city": "ATAPI ${params.atName} City",
        "district": "ATAPI ${params.atName} District",
        "urban": "ATAPI ${params.atName} Urban"
    },
    "users": {
        "fullname": "ATAPI ${params.atName} users fullname",
        "email":"ATAPI${params.atName}@atapi.com",
        "userCode": "ATAPI${params.atName}Usrs"
    },
    "usersSales": {
        "fullname": "ATAPI ${params.atName} users sales fullname",
        "email":"ATAPI${params.atName}@sales.com",
        "userCode": "ATAPI${params.atName}UsSl"
    },
    "vouchers": {
        "name": "ATAPI {params.atName} Name",
        "externalId": "ATAPI${params.atName}Vouc",
        "code": "${params.atName}",
        "description": "ATAPI ${params.atName} Description",
        "header":"ATAPI ${params.atName} Header",
        "terms": "{ATAPI ${params.atName} Terms and Condition}",
        "instructions": "{ATAPI ${params.atName} Instructions}",
        "tag": "{ATAPI ${params.atName} Tag}"
    },
    "warehouses": {
        "name": "ATAPI ${params.atName} Warehouse name",
        "code": "ATAPI${params.atName}WH",
        "externalId": "ATAPI${params.atName}WH",
        "noteAddress": "ATAPI ${params.atName} Note Address",
        "address": "ATAPI ${params.atName} Address"
    },
    "warehouseValues": {
        "name": "ATAPI ${params.atName} Warehouse Value name"
    },
    "warehouseTemperatures": {
        "name": "ATAPI ${params.atName} Warehouse Temperature name"
    }
}