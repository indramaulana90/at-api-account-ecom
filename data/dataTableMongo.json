[
	{
		"idInvoiceGroups": {
			"isMongo": true
		}
	},
	{
		"idInvoiceGroupBrands": {
			"isMongo": true
		}
	},
	{
		"idBrands": {
			"isMongo": true
		}
	},
	{
		"idProducts": {
			"isMongo": true
		}
	},
	{
		"idCart": {
			"isMongo": true
		}
	}
]
