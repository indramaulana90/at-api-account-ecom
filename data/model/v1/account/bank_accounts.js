/**
* Author      : Evinna Ginola Utama (evinna.ginola@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This File for Model Seeding Data on bank_accounts table
*/

import seedingData from '@at-api-core/seeding-data';

/**
 * This function for seeding data select by id on bank_accounts table.
 * @param {Number} id bank_accounts id.
 * @return {ArrayOfObject} result from db.
 */
export const selectById = async (id) => {
  const query = {
    type: 'select',
    table: 'public.bank_accounts',
    condition: { id },
    limit: 5,
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};

/**
 * This function for seeding data select by id on bank_accounts table.
 * @param {Number} userId user_id from bank_accounts table.
 * @return {ArrayOfObject} result from db.
 */
export const selectByUserId = async (userId) => {
  const query = {
    type: 'select',
    table: 'public.bank_accounts',
    condition: { user_id: userId },
    limit: 5,
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};

/**
 * This function for seeding data delete by id on bank_accounts table.
 * @param {Object} params object on will be inserted with:
 * @param {Number} userId user_id from bank_accounts table.
 * @param {Number} params.idBankAccounts id from bank_accounts table.
 */
export const deleteById = async (params) => {
  const query = {
    type: 'remove',
    table: 'public.bank_accounts',
    condition: {
      id: params.idBankAccounts,
    },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};

export const deleteByUserId = async (userId) => {
  const query = {
    type: 'remove',
    table: 'public.bank_accounts',
    condition: {
      user_id: userId,
    },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};
