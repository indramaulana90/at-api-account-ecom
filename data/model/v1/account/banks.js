/**
* Author      : Fauzan Novaldi Suteja (fauzan.novaldi@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This File for Model Seeding Data on banks table
*/

import seedingData from '@at-api-core/seeding-data';

/**
 * This function for seeding data select by id on banks table.
 * @param {Number} id banks id.
 * @return {ArrayOfObject} result from db.
 */
export const selectById = async (id) => {
  const query = {
    type: 'select',
    table: 'public.banks',
    condition: { id },
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};

/**
 * This function for seeding data delete by id on banks table.
 * @param {Object} params object on will be inserted with:
 * @param {Number} params.idBanks id from banks table.
 */
export const deleteById = async (params) => {
  const query = {
    type: 'remove',
    table: 'public.banks',
    condition: {
      id: params.idBanks,
    },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};
