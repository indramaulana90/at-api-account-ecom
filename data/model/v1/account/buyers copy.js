/**
* Author      : Evinna Ginola Utama (evinna.ginola@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This File for Model Seeding Data on buyers table
*/

import seedingData from '@at-api-core/seeding-data';

/**
 * This function for seeding data select by id on buyers table.
 * @param {Number} id buyers id.
 * @return {ArrayOfObject} result from db.
 */
export const selectById = async (id) => {
  const query = {
    type: 'select',
    table: 'public.buyers',
    condition: { id },
    limit: 5,
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};

/**
 * This function for seeding data reset account by id on users table.
 * @param {Number} id users id.
 */
export const resetAccountChimera = async (id) => {
  const query = {
    type: 'update',
    table: 'public.buyers',
    modifier: {
      location_id: '411780000000000000000000',
      vehicle_accessibility_id: 1,
      name: 'Chimera-SR',
      address: 'Jl. Kemang Sel. II, Kompleks Kiray No.34',
      note_address: 'notes address for ATAPI',
      tax_no: '123456789012345',
      longitude: 106.8130237,
      latitude: 6.2629731,
      phone_no: '0212251625',
      image_url: 'https://sinbad.co.id/image.jpg',
      tax_image_url: 'https://sinbad.co.id/image.jpg',
      vehicle_accessibility_amount: 2,
    },
    condition: { id },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};

/**
 * This function for seeding data delete by id on buyers table.
 * @param {Number} id id from buyers table.
 */
export const deleteById = async (id) => {
  const query = {
    type: 'remove',
    table: 'public.buyers',
    condition: { id },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};
