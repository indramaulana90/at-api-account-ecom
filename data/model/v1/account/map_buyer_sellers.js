/**
* Author      : Evinna Ginola Utama (evinna.ginola@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This File for Model Seeding Data on map_buyer_sellers table
*/

import seedingData from '@at-api-core/seeding-data';
import getModelValues from '@at-api-core/get-model-values';

/**
 * This function for seeding data select by id on map_buyer_sellers table.
 * @param {Number} id map_buyer_sellers id.
 * @return {ArrayOfObject} result from db.
 */
export const selectById = async (id) => {
  const query = {
    type: 'select',
    table: 'public.map_buyer_sellers',
    condition: { id },
    limit: 5,
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};

/**
 * This function for seeding data select by id on map_buyer_sellers table.
 * @param {Number} buyerId map_buyer_sellers id.
 * @return {ArrayOfObject} result from db.
 */
export const selectByBuyerId = async (buyerId) => {
  const query = {
    type: 'select',
    table: 'public.map_buyer_sellers',
    condition: { buyer_id: buyerId },
    limit: 5,
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};

/**
 * This function for seeding data insert with half params on map_buyer_sellers table.
 * @param {Object} params object on will be inserted with:
 * @param {Number} params.idMapBuyerSellers id from map_buyer_sellers table.
 * @param {String} params.atName Original name of AT Name without space maximum 11 char (e.g. StoreDtail).
 */
export const insertHalfParams = async (params) => {
  const query = {
    type: 'insert',
    table: 'public.map_buyer_sellers',
    values: {
      id: params.idMapBuyerSellers,
      external_id: getModelValues('mapBuyerSellers', params).externalId,
      is_bank_account_verified: false,
      bank_id: null,
    },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};

/**
 * This function for seeding data insert with half params on map_buyer_sellers table.
 * @param {Object} params object on will be inserted with:
 * @param {Number} params.idMapBuyerSellers id from map_buyer_sellers table.
 * @param {String} params.atName Original name of AT Name without space maximum 11 char (e.g. StoreDtail).
 */
export const insertFullParams = async (params) => {
  const query = {
    type: 'insert',
    table: 'public.map_buyer_sellers',
    values: {
      id: params.idMapBuyerSellers,
      external_id: getModelValues('mapBuyerSellers', params).externalId,
      name: getModelValues('mapBuyerSellers', params).name,
    },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};

/**
 * This function for seeding data delete by id on map_buyer_sellers table.
 * @param {Object} params object on will be inserted with:
 * @param {Number} params.idMapBuyerSellers id from map_buyer_sellers table.
 */
export const deleteById = async (params) => {
  const query = {
    type: 'remove',
    table: 'public.map_buyer_sellers',
    condition: {
      id: params.idMapBuyerSellers,
    },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};
