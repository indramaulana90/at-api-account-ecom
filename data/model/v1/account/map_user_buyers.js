/**
* Author      : Evinna Ginola Utama (evinna.ginola@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This File for Model Seeding Data on map_user_buyers table
*/

import seedingData from '@at-api-core/seeding-data';

/**
 * This function for seeding data select by id on map_user_buyers table.
 * @param {Number} id map_user_buyers id.
 * @return {ArrayOfObject} result from db.
 */
export const selectById = async (id) => {
  const query = {
    type: 'select',
    table: 'public.map_user_buyers',
    condition: { id },
    limit: 5,
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};

/**
 * This function for seeding data select by user_id on map_user_buyers table.
 * @param {Number} userId map_user_buyers id.
 * @return {ArrayOfObject} result from db.
 */
export const selectByUserId = async (userId) => {
  const query = {
    type: 'select',
    table: 'public.map_user_buyers',
    condition: { user_id: userId },
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};

/**
 * This function for seeding data delete by user_id on map_user_buyers table.
 * @param {Number} userId user_id from map_user_buyers table.
 */
export const deleteByUserId = async (userId) => {
  const query = {
    type: 'remove',
    table: 'public.map_user_buyers',
    condition: {
      user_id: userId,
    },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};

/**
 * This function for seeding data reset status of sompleted data on map_user_buyers table.
 * @param {Number} userId user_id from map_user_buyers table.
 */
export const resetCompletedData = async (userId) => {
  const query = {
    type: 'update',
    table: 'public.map_user_buyers',
    modifier: { is_data_completed: false },
    condition: {
      user_id: userId,
    },
  };

  await seedingData(query, 'ms_ng_ecom_account');
};
