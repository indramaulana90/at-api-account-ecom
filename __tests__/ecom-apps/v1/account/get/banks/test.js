/**
* Author      : Fauzan Novaldi Suteja (fauzan.novaldi@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This Main File for Testing on API get_banks
*/

import getDataFromPool from '@at-api-core/get-data-from-pool';
import * as er from '@at-api-core/expect-response';
import endpoint from './endpoint';
import * as data from './data';
import tc from './tc';
import schema from './schema.json';
import rs from '../../../../../../helper/responseStatusLibrary.json';
import { selectById as sdSelectBanks } from '../../../../../../data/model/v1/account/banks';

const { cookies } = getDataFromPool('ng','sinbad-app', 'Chimera-SR_081231111111');

describe(`${tc.type[1]} ${tc.describe}`, () => {
  test(`${tc.positive.getAll}`, async () => {
    const queryParam = data.payloadEmpty();
    resp = await endpoint(cookies, queryParam);
    er.expectStatus(resp, rs.statusOk);
    er.expectSchema(resp, schema);

    const objectFiltered = resp.body.data[0];
    const objectDB = (await sdSelectBanks(objectFiltered.id))[0];
    er.expectEqualDB(resp, objectFiltered, objectDB, data.compareDB);
  });

  data.loopingParameter.forEach(async (element) => {
    test(`${element.title}`, async () => {
      const payload = data.payloadEmpty();
      payload[Object.keys(element.value)] = Object.values(element.value)[0];
      resp = await endpoint(cookies, payload);
      er.expectStatus(resp, rs.statusOk);
      er.expectSchema(resp, schema);

      const objectFiltered = resp.body.data[0];
      const objectDB = (await sdSelectBanks(objectFiltered.id))[0];
      er.expectEqualDB(resp, objectFiltered, objectDB, data.compareDB);
    });
  });

  data.loopingParameterEmptyNull.forEach(async (element) => {
    test(`${element.title}`, async () => {
      const payload = data.payloadEmpty();
      payload[Object.keys(element.value)] = Object.values(element.value)[0];
      resp = await endpoint(cookies, payload);
      er.expectStatus(resp, rs.statusOk);
      er.expectSchema(resp, schema);
      er.expectSnapshot(resp);
    });
  });

  test(`${tc.positive.getWithXUserId}`, async () => {
    const queryParam = data.payloadEmpty();
    resp = await endpoint(cookies, queryParam).set('x-user-id', 1);
    er.expectStatus(resp, rs.statusOk);
    er.expectSchema(resp, schema);

    const objectFiltered = resp.body.data[0];
    const objectDB = (await sdSelectBanks(objectFiltered.id))[0];
    er.expectEqualDB(resp, objectFiltered, objectDB, data.compareDB);
  });

  test(`${tc.positive.getWithXUserId}`, async () => {
    const queryParam = data.payloadEmpty();
    resp = await endpoint(cookies, queryParam);
    er.expectStatus(resp, rs.statusOk);
    er.expectSchema(resp, schema);

    const objectFiltered = resp.body.data[0];
    const objectDB = (await sdSelectBanks(objectFiltered.id))[0];
    er.expectEqualDB(resp, objectFiltered, objectDB, data.compareDB);
  });
});

describe(`${tc.type[0]} ${tc.describe}`, () => {
  data.loopingParameterInvalid.forEach(async (element) => {
    test(`${element.title}`, async () => {
      const payload = data.payloadEmpty();
      payload[Object.keys(element.value)] = Object.values(element.value)[0];
      resp = await endpoint(cookies, payload);
      er.expectStatus(resp, rs.statusOk);
      er.expectSchema(resp, schema);
      er.expectSnapshot(resp);
    });
  });

  test(`${tc.negative.paramNotSet}`, async () => {
    const queryParam = data.payloadEmpty();
    resp = await endpoint(cookies, queryParam);
    er.expectStatus(resp, rs.statusOk);
    er.expectSchema(resp, schema);

    const objectFiltered = resp.body.data[0];
    const objectDB = (await sdSelectBanks(objectFiltered.id))[0];
    er.expectEqualDB(resp, objectFiltered, objectDB, data.compareDB);
  });
});

describe(`${tc.type[0]} ${tc.describe}`, () => {
  test(`${tc.negative.xPlatformNull}`, async () => {
    const payload = data.payloadEmpty();

    resp = await endpoint(cookies, payload).set('x-platform', null);
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
    er.expectSnapshot(resp);
  });

  test(`${tc.negative.xPlatformEmpty}`, async () => {
    const payload = data.payloadEmpty();

    resp = await endpoint(cookies, payload).set('x-platform', '');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
    er.expectSnapshot(resp);
  });

  test(`${tc.negative.xPlatformInvalid}`, async () => {
    const payload = data.payloadEmpty();

    resp = await endpoint(cookies, payload).set('x-platform', 'agent-app');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
    er.expectSnapshot(resp);
  });

  test(`${tc.negative.xPlatformNotSet}`, async () => {
    const payload = data.payloadEmpty();

    resp = await endpoint(cookies, payload).unset('x-platform');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
    er.expectSnapshot(resp);
  });

  data.loopingErrorCookies.forEach(async (element) => {
    test(`${element.title}`, async () => {
      const payload = data.payloadEmpty();
      const cookiesNegative = element.value;

      resp = await endpoint(cookiesNegative, payload);
      er.expectStatus(resp, rs.statusUnauthorized);
      er.expectUnauthorized(resp);
      er.expectSnapshot(resp);
    });
  });
});
