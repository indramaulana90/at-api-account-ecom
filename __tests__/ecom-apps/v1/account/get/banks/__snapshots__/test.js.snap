// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@negative get banks -- As a Store, I should not be able to get all list because cookies set to Empty 1`] = `Object {}`;

exports[`@negative get banks -- As a Store, I should not be able to get all list because cookies set to Invalid 1`] = `Object {}`;

exports[`@negative get banks -- As a Store, I should not be able to get all list because cookies set to Null 1`] = `Object {}`;

exports[`@negative get banks -- As a Store, I should not be able to get all list because x-platform header is not set 1`] = `Object {}`;

exports[`@negative get banks -- As a Store, I should not be able to get all list because x-platform invalid 1`] = `Object {}`;

exports[`@negative get banks -- As a Store, I should not be able to get all list of bank with parameter integer 1`] = `
Object {
  "data": Array [],
}
`;

exports[`@negative get banks -- As a Store, I should not be able to get all list of bank with parameter invalid data 1`] = `
Object {
  "data": Array [],
}
`;

exports[`@negative get banks -- As a Store, I should not be able to get all list of bank with x-platform empty 1`] = `Object {}`;

exports[`@negative get banks -- As a Store, I should not be able to get all list of bank with x-platform null 1`] = `Object {}`;

exports[`@positive get banks -- As a Store, I should be able to get all list of bank with parameter empty 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "name": "Bank BCA",
    },
    Object {
      "id": 2,
      "name": "Bank BNI",
    },
    Object {
      "id": 4,
      "name": "Bank BRI",
    },
    Object {
      "id": 3,
      "name": "Bank Mandiri",
    },
  ],
}
`;

exports[`@positive get banks -- As a Store, I should be able to get all list of bank with parameter null 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "name": "Bank BCA",
    },
    Object {
      "id": 2,
      "name": "Bank BNI",
    },
    Object {
      "id": 4,
      "name": "Bank BRI",
    },
    Object {
      "id": 3,
      "name": "Bank Mandiri",
    },
  ],
}
`;
